### PR TITLE
Revamp gRPC test server for api tests

### PR DESCRIPTION
**Pull request by azdagron**
Unifies server implementation. The test server ensures that all handlers invocations are done when the server is cleaned up. This allows tests that want to check post-streaming RPC conditions deterministically.

### CODE REVIEW

File: pkg/agent/endpoints/workload/handler_test.go [hunk 1]
```diff
@@ -1508,45 +1508,31 @@ func runTest(t *testing.T, params testParams, fn func(ctx context.Context, clien
...
-	server := grpc.NewServer(
-		grpc.UnaryInterceptor(unaryInterceptor),
-		grpc.StreamInterceptor(streamInterceptor),
+	), grpctest.OverUDS(),
 	)
-	workloadPB.RegisterSpiffeWorkloadAPIServer(server, handler)
-	addr := spiretest.ServeGRPCServerOnTempUDSSocket(t, server)
+
+	//func(s grpc.ServiceRegistrar, srv "github.com/spiffe/go-spiffe/v2/proto/spiffe/workload".SpiffeWorkloadAPIServer)
+	//func(*grpc.Server, "github.com/spiffe/go-spiffe/v2/proto/spiffe/workload".SpiffeWorkloadAPIServer)
```
Comment (MarcosDY) [inline pkg/agent/endpoints/workload/handler_test.go]: leftover comments?

File: test/grpctest/server.go [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+	errCh := make(chan error, 1)
+	go func() {
+		errCh <- server.Serve(serverListener)
+	}()
+
+	var stopOnce sync.Once
+	stop := func() {
+		stopOnce.Do(func() {
+			defer func() {
+				tb.Logf("Waiting for handlers to drain")
```
Comment (MarcosDY) [inline test/grpctest/server.go:119]: tb.Log?

**Review by MarcosDY — APPROVED** [review]
LGTM, 
Just a minor comment that you can avoid

**Comment by rturner3** [comment]
Merging this to hopefully make CI a little more stable, hope you're ok with comments being handled in another PR @MarcosDY

**Pull request merged**